api/src/main/java/org/datacleaner/descriptors/DescriptorProvider.java

```diff
@@ -198,4 +198,17 @@
      * @param listener
      */
     public void removeListener(DescriptorProviderListener listener);
+
+    /**
+     * Get information about providers.
+     *
+     * @return
+     */
+    public Map<DescriptorProvider, DescriptorProviderStatus> getActualStatusMap();
```

Comment:
Hmm, to be honest I'm quite concerned about the changes here.

Most importantly on the return type: The fact that it returns a Map is really leaking the fact that this was made for a composite implementation.

Secondly: Why is this even needed? I thought this was exactly what the listeners are for - so the information can be given to the listeners instead of them having to pull it out themselves.

Thirdly: The naming - "Actual" - this seems to indicate to me that there is something "unactual" in play also, or?